### PR TITLE
Remove tokio::spawn from HashAggregateExec (#2201)

### PR DESCRIPTION
**Pull request by tustvold**
# Which issue does this PR close?

Part of #2201

 # Rationale for this change

See ticket

# What changes are included in this PR?

Much like #2203 removed the `tokio::spawn` from WindowAggExec, this removes the `tokio::spawn` from HashAggregateExec 

# Are there any user-facing changes?

No

### CODE REVIEW

File: datafusion/core/src/physical_plan/hash_aggregate.rs [hunk 1]
```diff
@@ -356,25 +354,130 @@ Example: average
...
+    }
+}
+
+impl RecordBatchStream for GroupedHashAggregateStream {
+    fn schema(&self) -> SchemaRef {
+        self.schema.clone()
     }
 }
 
+/// TODO: Make this a member function of [`GroupedHashAggregateStream`]
```
Comment (tustvold) [inline datafusion/core/src/physical_plan/hash_aggregate.rs:471]: I left this as a TODO for a better diff, I'll follow up with a subsequent PR doing just this

File: datafusion/core/src/physical_plan/hash_aggregate.rs [hunk 2]
```diff
@@ -798,35 +751,23 @@ impl HashAggregateStream {
         aggr_expr: Vec<Arc<dyn AggregateExpr>>,
         input: SendableRecordBatchStream,
         baseline_metrics: BaselineMetrics,
-    ) -> Self {
-        let (tx, rx) = futures::channel::oneshot::channel();
-
-        let schema_clone = schema.clone();
-        let elapsed_compute = baseline_metrics.elapsed_compute().clone();
-        let join_handle = tokio::spawn(async move {
```
Comment (tustvold) [inline datafusion/core/src/physical_plan/hash_aggregate.rs:806]: This logic is moved into the Stream::poll_next

File: datafusion/core/src/physical_plan/hash_aggregate.rs [hunk 3]
```diff
@@ -748,46 +732,15 @@ fn aggregate_expressions(
...
-        schema: SchemaRef,
-        #[pin]
-        output: futures::channel::oneshot::Receiver<ArrowResult<RecordBatch>>,
-        finished: bool,
-        drop_helper: AbortOnDropSingle<()>,
-    }
-}
-
-/// Special case aggregate with no groups
-async fn compute_hash_aggregate(
```
Comment (tustvold) [inline datafusion/core/src/physical_plan/hash_aggregate.rs:763]: This logic is moved into `HashAggregateStream::poll_next`

File: datafusion/core/src/physical_plan/hash_aggregate.rs [hunk 4]
```diff
@@ -516,88 +619,7 @@ fn group_aggregate_batch(
                 })
         })?;
 
-    Ok(accumulators)
-}
-
-async fn compute_grouped_hash_aggregate(
```
Comment (tustvold) [inline datafusion/core/src/physical_plan/hash_aggregate.rs:522]: This logic is moved into a combination of `GroupedHashAggregateStream::new` and `GroupedHashAggregateStream::poll_next`

File: datafusion/core/src/physical_plan/hash_aggregate.rs [hunk 5]
```diff
@@ -748,46 +732,15 @@ fn aggregate_expressions(
...
-        finished: bool,
-        drop_helper: AbortOnDropSingle<()>,
-    }
-}
-
-/// Special case aggregate with no groups
-async fn compute_hash_aggregate(
-    mode: AggregateMode,
+/// stream struct for hash aggregation
+pub struct HashAggregateStream {
```
Comment (yjshen) [inline datafusion/core/src/physical_plan/hash_aggregate.rs:736]: `Hash` makes no sense each time I saw it 🙃

**Review by yjshen — APPROVED** [review]

**Review by alamb — COMMENTED** [review]
looks cool -- it might be worth running the tpch benchmarks at some point to make sure we didn't inadvertently slow things down

**Pull request merged**